WordPress/openverse-catalog
# Extract media type from staged tsv file name for loader

`Tsv_to_postgres_loader` DAG handles getting data from TSV files, saving it to S3 and intermediate database, cleaning it and finally saving the clean data to the main database.

This PR adds media type (currently `image`, and in the future, `audio`) to the tsv filename. The media type is used in the downstream tasks to use the correct functions/columns when loading the data into the database. 

Different tasks of the `tsv_to_postgres_loader` DAG pass the media_type data either using the filename, or, if it is not available, using the Airflow xcom messaging (xcom.push/xcom.pull).

To test this PR you can:
1. build the Docker containers and run all tests.
2. build the Docker containers and run one of the image ingestion DAGS (Flickr workflow (not the Flickr ingestion workflow!) runs comparatively fast, but it needs authentication, Science museum workflow doesn't require authentication). After the ingestion workflow is complete, turn on the `tsv_to_postgres_loader`. On its completion, you will be able to see the ingested data in openledger database.

Signed-off-by: Olga Bulat <obulat@gmail.com>Signed-off-by: Olga Bulat <obulat@gmail.com>
